src/map/atcommand.c

```diff
@@ -7253,12 +7253,16 @@ ACMD(mobinfo)
 			}
 #endif
 
-			if (item_data->slot)
-				snprintf(atcmd_output2, sizeof(atcmd_output2), " - %s[%d]  %02.02f%%", item_data->jname, item_data->slot, (float)droprate / 100);
-			else
-				snprintf(atcmd_output2, sizeof(atcmd_output2), " - %s  %02.02f%%", item_data->jname, (float)droprate / 100);
-
-			strcat(atcmd_output, atcmd_output2);
+			struct item link_item = { 0 };
+			link_item.nameid = monster->dropitem[i].nameid;
+			StringBuf buf;
+			StrBuf->Init(&buf);
+			StrBuf->AppendStr(&buf, " - ");
+			clif->format_itemlink(&buf, &link_item);
+			snprintf(atcmd_output2, sizeof(atcmd_output2), " %02.02f%%", (float)droprate / 100);
+			StrBuf->AppendStr(&buf, atcmd_output2);
```

Comment:
This copy can be skipped by formatting the string directly into the stringbuf

```suggestion
			StrBuf->Printf(buf, " %02.02f%%", (float)droprate / 100);
```

Reply:
This is now changed.